src/algebra/fract.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2019 Kevin Buzzard. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard
+
+Fractional parts of reals (and more general floor_rings)
+-/
+
+-- note heavy imports. Is this bad? This is why I made a separate file
+-- rather than dropping into archimedean.lean
+import algebra.archimedean tactic.linarith tactic.abel
+
+variables {α : Type*} [linear_ordered_ring α] [floor_ring α]
+
+/-- The fractional part fract r of r is just r - ⌊r⌋ -/
+def fract (r : α) : α := r - ⌊r⌋
+
+-- Mathematical notation is usually {r}. Let's not even go there.
+-- I have no notation currently.
+
+-- My mathematical instinct tells me that the order is ⌊r⌋ + {r}
+-- but then I have to cast in the statement of the theorem. Should I care?
+@[simp] theorem fract_add_floor (r : α) : fract r + ⌊r⌋ = r := sub_add_cancel _ _
+
+theorem fract_nonneg (r : α) : 0 ≤ fract r :=
+sub_nonneg.2 $ floor_le _
+
+theorem fract_lt_one (r : α) : fract r < 1 :=
+sub_lt.1 $ sub_one_lt_floor _
+
+@[simp] lemma fract_zero : fract (0 : α) = 0 := by unfold fract;simp
+
```

Comment:
I would also expect two simp-lemmas `fract_floor` and `floor_fract`.

Reply:
`fract_floor`: do you want me to prove `fract (floor r) = 0` or `fract ((floor r) : alpha) = 0`? I went for the latter. I did `floor_fract` and threw in `fract_fract` for good measure.

I put spaces after , and ;